presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -311,7 +311,7 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
             }
```

Comment:
This commit (in isolation) causes some unit tests to fail. For example, incorrectly complains that this query has a correlated subquery:

```sql
SELECT o1.orderkey, COUNT(*) 
FROM orders o1 LEFT OUTER JOIN (SELECT * FROM orders LIMIT 10) o2 ON EXISTS(SELECT avg(orderkey) FROM orders) 
GROUP BY o1.orderkey
```